dm/config/task_test.go

```diff
@@ -192,4 +202,67 @@ syncers:
 	c.Assert(taskConfig.MySQLInstances[2].Mydumper.Threads, Equals, 44)
 	c.Assert(taskConfig.MySQLInstances[2].Loader.PoolSize, Equals, 55)
 	c.Assert(taskConfig.MySQLInstances[2].Syncer.WorkerCount, Equals, 66)
+
+	configContent = []byte(`---
+name: test
+task-mode: all
+is-sharding: false
+meta-schema: "dm_meta"
+remove-meta: false
+enable-heartbeat: true
+heartbeat-update-interval: 1
+heartbeat-report-interval: 1
+timezone: "Asia/Shanghai"
+
+target-database:
+  host: "127.0.0.1"
+  port: 4000
+  user: "root"
+  password: ""
+
+mysql-instances:
+  - source-id: "mysql-replica-01"
+  - source-id: "mysql-replica-02"
+  - source-id: "mysql-replica-03"
+
+black-white-list:
+  instance:
+    do-dbs: ["test"]
+
+routes: 
+  route-rule-1:
+  route-rule-2:
+  route-rule-3:
+  route-rule-4:
+
+filters:
+  filter-rule-1:
+  filter-rule-2:
+  filter-rule-3:
+  filter-rule-4:
+`)
+
+	err = ioutil.WriteFile(filepath, configContent, 0644)
+	c.Assert(err, IsNil)
+	taskConfig = NewTaskConfig()
+	err = taskConfig.DecodeFile(filepath)
+	c.Assert(err, IsNil)
+	taskConfig.MySQLInstances[0].RouteRules = []string{"route-rule-1", "route-rule-2", "route-rule-1", "route-rule-2"}
+	taskConfig.MySQLInstances[1].FilterRules = []string{"filter-rule-1", "filter-rule-2", "filter-rule-3", "filter-rule-2"}
+	err = taskConfig.adjust()
+	c.Assert(terror.ErrConfigDuplicateCfgItem.Equal(err), IsTrue)
+	c.Assert(err, ErrorMatches, "[\\s\\S]*mysql-instance\\(0\\)'s route-rules: route-rule-1, route-rule-2[\\s\\S]*")
```

Comment:
you can use ``` ` ``` instead of ` " `to wrap a literal string, then it can be more readable.